src/scenic/core/object_types.py

```diff
@@ -109,6 +109,8 @@ def __init_subclass__(cls):
                 ty = underlyingType(getattr(inst, prop))
             dynTypes[prop] = ty
         cls._dynamicProperties = dynTypes
+        cls._dynamicallyUpdatedProperties = {prop: val for prop, val
```

Comment:
Let's give this a more informative name: all dynamic properties are updated every time step, even `final` ones, so this doesn't really distinguish them. How about something like `_simulatorProvidedProperties`?